homeassistant/components/devolo_home_control/sensor.py

```diff
@@ -0,0 +1,153 @@
+"""Platform for sensor integration."""
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.components.sensor import (
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_ILLUMINANCE,
+    DEVICE_CLASS_TEMPERATURE,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import DOMAIN
+from .devolo_device import DevoloDeviceEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Get all binary sensor and multi level sensor devices and setup them via config entry."""
+    entities = []
+
+    for device in hass.data[DOMAIN]["homecontrol"].binary_sensor_devices:
+        for binary_sensor in device.binary_sensor_property:
+            entities.append(
+                DevoloBinaryDeviceEntity(
+                    homecontrol=hass.data[DOMAIN]["homecontrol"],
+                    device_instance=device,
+                    element_uid=binary_sensor,
+                )
+            )
```

Comment:
Binary sensors should be in `binary_sensor.py`

Reply:
Please remove them from this PR, and do another pr for binary sensors after this one merges

Reply:
This is done with the latest commit